### ISSUE TITLE
Huge memory usage

### ISSUE THREAD

**Issue by mrdc** [sienori/Tab-Session-Manager#881]
## Short description
I have 545 sessions in TSM (some of them containing multiple windows) and memory usage is high - TSM extension uses 7700 MB (see details below).

## Steps to reproduce

1. Start browser with clean profile
2. Install Tab Session Manager
3. Save a lot of sessions
4. Check memory usage by extension

## Expected result
TSM should use less memory.

## Actual result
TSM uses a log of memory:
![TSM_mem](https://user-images.githubusercontent.com/16876990/135304527-06fb9368-6f61-410f-a87c-0b5fc062c333.png)

![tsm-mem](https://user-images.githubusercontent.com/16876990/135304558-082f7c3f-adb5-4315-a9aa-897b044cc2c8.png)

## Platform information
* Platform (OS): macOS 10.15
* Version of browser: Firefox 92.0.1 x64
* Version of Tab Session Manager: Version 6.10.1

**Comment by PaddlesAU** [comment]
I'm not sure if it's related, but I've noticed significant memory usage lately too. I also discovered that I had TSM's "Startup - Open previous session" _and_ Firefox's "Startup - Restore previous session" selected. When I switched off the standard Firefox setting and restarted, initial RAM use by Firefox was 3-4GB lower. My normal startup session has 50-60 tabs in one Window.
OS: Windows 10 Pro x64 build 10.0.19043; Firefox 92.0.1 x64; TSM: 6.10.1
